### ISSUE TITLE
New Rule Proposal: Disallow Undefined Methods

### ISSUE THREAD

**Issue by TheSavior** [yannickcr/eslint-plugin-react#337]
A component like this:

```
var React = require('react');

var MyComponent = React.createClass({
  render: function() {
    return <span onClick={this.handleClick}>Span</span>;
  }
});

module.exports = MyComponent;
```

Should complain because the component doesn't have a `handleClick` function.

**Comment by lencioni** [comment]
I wonder if a more general rule for this could be written.
